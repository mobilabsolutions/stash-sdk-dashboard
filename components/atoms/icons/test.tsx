/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'

import { withTestSetup } from '../../../test_utils'
import Logo from './logo'
import AccountIcon from './account'
import KeyIcon from './key'
import CopyIcon from './copy'
import DownIcon from './down'
import UpIcon from './up'

const TestLogo = withTestSetup(Logo)
const TestAccountIcon = withTestSetup(AccountIcon)
const TestKeyIcon = withTestSetup(KeyIcon)
const TestCopyIcon = withTestSetup(CopyIcon)
const TestDownIcon = withTestSetup(DownIcon)
const TestUpIcon = withTestSetup(UpIcon)

it('Logo Should Render', () => {
  const component = renderer.create(<TestLogo />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('AccountIcon Should Render', () => {
  const component = renderer.create(<TestAccountIcon />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('KeyIcon Should Render', () => {
  const component = renderer.create(<TestKeyIcon />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('CopyIcon Should Render', () => {
  const component = renderer.create(<TestCopyIcon />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('UpIconIcon Should Render', () => {
  const component = renderer.create(<TestUpIcon />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('DownIcon Should Render', () => {
  const component = renderer.create(<TestDownIcon />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
