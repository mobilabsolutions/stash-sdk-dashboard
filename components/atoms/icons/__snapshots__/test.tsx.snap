// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountIcon Should Render 1`] = `
<svg
  className="enable"
  height={20}
  onClick={[Function]}
  viewBox="0 0 20 20"
  width={20}
>
  <path
    d="M10 0C4.48 0 0 4.48 0 10s4.48 10 10 10 10-4.48 10-10S15.52 0 10 0zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2a7.2 7.2 0 0 1-6-3.22c.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08a7.2 7.2 0 0 1-6 3.22z"
    fill="#D6D9DB"
    fillRule="evenodd"
  />
</svg>
`;

exports[`CopyIcon Should Render 1`] = `
<svg
  className="enable"
  height={24}
  onClick={[Function]}
  viewBox="0 0 24 24"
  width={24}
>
  <g
    fill="none"
    fillRule="evenodd"
    stroke="#D4D4D4"
    strokeWidth="2"
  >
    <rect
      height="12"
      rx="2"
      width="12"
      x="3"
      y="4"
    />
    <path
      d="M15.119 8L8 16.32V19a1 1 0 0 0 1 1h10a1 1 0 0 0 1-1V9a1 1 0 0 0-1-1h-3.881z"
    />
  </g>
</svg>
`;

exports[`DownIcon Should Render 1`] = `
<svg
  className="enable"
  height={20}
  onClick={[Function]}
  viewBox="0 0 48 48"
  width={20}
>
  <path
    d="M14.83 16.42L24 25.59l9.17-9.17L36 19.25l-12 12-12-12z"
  />
</svg>
`;

exports[`Illustration Should Render 1`] = `
<svg
  className="enable"
  height={279}
  onClick={[Function]}
  viewBox="0 0 359 279"
  width={359}
>
  <defs>
    <linearGradient
      id="a"
      x1="97.659%"
      x2="0%"
      y1="67.527%"
      y2="67.527%"
    >
      <stop
        offset="0%"
        stopColor="#F5F5F5"
      />
      <stop
        offset="100%"
        stopColor="#D4D6E5"
      />
    </linearGradient>
    <linearGradient
      id="b"
      x1="67.053%"
      x2="39.067%"
      y1="18.447%"
      y2="83.535%"
    >
      <stop
        offset="0%"
        stopColor="#4F5995"
      />
      <stop
        offset="100%"
        stopColor="#131E61"
      />
    </linearGradient>
    <path
      d="M.038.052h8.907v8.896H.038z"
      id="c"
    />
    <path
      d="M.038.052h8.907v8.896H.038z"
      id="e"
    />
  </defs>
  <g
    fill="none"
    fillRule="evenodd"
  >
    <path
      d="M165.585 270.193c90.385 0 193.397 37.913 193.397-44.325 0-33.36-47.011-72.711-64.1-136.45C283.596 47.333 250.884 0 197.162 0 143.861 0 95.302 46.368 65.42 82.259 44.63 107.229 0 173.027 0 206.767c0 82.239 75.2 63.426 165.585 63.426z"
      fill="#FFF"
    />
    <g
      transform="translate(31 56)"
    >
      <g
        transform="rotate(-19 340.272 -60.733)"
      >
        <rect
          fill="#EDEEF5"
          height="84.822"
          rx="9.216"
          width="126.536"
        />
        <path
          d="M8.252 59.996h44.7v4.827h-44.7zM8.252 51.031h30.946v4.827H8.252zM8.252 68.961h61.205v4.827H8.252z"
          fill="#F9C2BA"
        />
      </g>
      <path
        d="M.657 56.342h125.386v17.194H.657z"
        fill="url(#a)"
        transform="rotate(-19 216.192 -39.272)"
      />
      <g
        transform="rotate(-19 181.569 -249.97)"
      >
        <rect
          fill="#D4D6E5"
          height="84.822"
          rx="9.216"
          width="126.536"
        />
        <ellipse
          cx="98.684"
          cy="70.685"
          fill="#ED4E39"
          rx="7.221"
          ry="7.241"
        />
        <ellipse
          cx="110.375"
          cy="70.685"
          fill="#ED4E39"
          rx="7.221"
          ry="7.241"
        />
        <path
          d="M9.791 13.725h25.445v21H9.791z"
          fill="#ED4E39"
        />
      </g>
      <rect
        fill="#D8D8D8"
        height="4.585"
        rx="2.292"
        width="203.608"
        x="68.203"
        y="195.368"
      />
      <rect
        fill="#D8D8D8"
        height="4.585"
        rx="2.292"
        width="26.458"
        x="276.412"
        y="195.368"
      />
      <rect
        fill="#D8D8D8"
        height="4.585"
        rx="2.292"
        width="10.353"
        x="263.758"
        y="195.368"
      />
      <rect
        fill="#D8D8D8"
        height="4.585"
        rx="2.292"
        width="10.353"
        x="307.471"
        y="195.368"
      />
      <path
        d="M89.314 0C62.63 0 41 21.554 41 48.142M89.314 16.047c-16.518 0-29.909 13.343-29.909 29.803M89.314 30.949c-8.26 0-14.955 6.671-14.955 14.9"
        stroke="#D4D6E5"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="3.456"
      />
      <g
        transform="translate(180.79 163.273)"
      >
        <path
          d="M3.516 3.439h26.678l3.165 28.656H0z"
          fill="#EDEEF5"
        />
        <path
          d="M25.451 8.597C25.451 3.849 21.588 0 16.824 0c-4.765 0-8.628 3.849-8.628 8.597"
          stroke="#F9C2BA"
          strokeWidth="2.304"
        />
        <rect
          fill="#D8D8D8"
          height="2.292"
          rx="1.146"
          width="4.601"
          x="5.895"
          y="8.024"
        />
        <rect
          fill="#D8D8D8"
          height="2.292"
          rx="1.146"
          width="4.601"
          x="23.15"
          y="8.024"
        />
      </g>
      <path
        d="M156.778 180.466h24.157v14.901h-24.157z"
        fill="#EDEEF5"
      />
      <path
        d="M167.13 180.466h4.602v3.44l-.888-.654-.626.653-.787-.653-.723.653-.665-.653-.912.653zM171.732 195.368h-4.601v-3.44l.888.654.626-.653.786.653.723-.653.666.653.912-.653z"
        fill="#F9C2BA"
      />
      <path
        d="M16.233 46.079C11.838 44.054-4.935 35.464 1.42 28.463c1.957-2.157 6.211 3.978 8.715 5.782 1.289.93 2.76.587 2.76-.977 0-3.452-10.31-5.933-12.138-13.096-.614-2.406 1.592-3.387 2.345-3.606 4.505-1.31 6.782 6.527 8.356 9.278.76 1.327 2.236 1.854 2.885.573 1.095-2.162-.393-4.566-2.416-7.21-2.604-3.404-10.035-5.86-6.094-11.404.96-1.35 3.094-1.72 4.302-.706 2.76 2.318 2.646 6.11 3.78 4.365 2.29-3.523-5.961-15.122 5.103-10.31 7.124 3.097-3.587 11.521-1.253 13.387 5.037 4.026 2.388-10.09 7.762-5.124 7.82 7.229-8.126 7.819-6.509 13.545.7 2.479 11.668-8 10.567-.141-.838 5.984-11.82 3.094-9.295 9.231 2.218 5.39 10.433-9.985 13.877-2.116.655 1.496.264 3.364-.299 4.898-.593 1.616-1.75 3.028-3.03 4.172-1.812 1.621-3.877 3.012-6.096 3.99-2.198.97-4.651 1.183-6.977 1.774l-1.533 1.31z"
        fill="url(#b)"
        transform="translate(220.046 125.218)"
      />
      <path
        d="M236.87 167.435c-1-4.03-1.047-8.314-.145-12.851.903-4.538.249-9.51-1.964-14.917"
        stroke="#090E2D"
        strokeLinecap="round"
        strokeWidth="1.152"
      />
      <g
        transform="translate(220.19 166.711)"
      >
        <path
          d="M29.843 28.656H3.166L0 0h33.36z"
          fill="#D4D6E5"
        />
        <rect
          fill="#D8D8D8"
          height="2.292"
          rx="1.146"
          width="2.301"
          x="25.163"
          y="24.071"
        />
      </g>
      <path
        d="M55 75.322h-3.32V72h-2.358v3.322H46v2.357h3.322V81h2.358v-3.32H55zM248 40.322h-3.32V37h-2.358v3.322H239v2.357h3.322V46h2.358v-3.32H248zM229.711 4.211l-8.736 2.496.528 1.835 8.736-2.496-.806 2.927 1.945.435 1.398-5.08-3.887-3.57-1.419 1.397zM18.393 109.33l-8.24 3.83.81 1.729 8.24-3.83-.339 3.017 1.99.126.585-5.235-4.4-2.917-1.182 1.6z"
        fill="#D8D8D8"
        fillRule="nonzero"
      />
      <g>
        <g
          transform="translate(211 100)"
        >
          <mask
            fill="#fff"
            id="d"
          >
            <use
              xlinkHref="#c"
            />
          </mask>
          <path
            d="M4.49 3.018a1.483 1.483 0 1 0 .002 2.966 1.483 1.483 0 0 0-.002-2.966m0 5.93A4.457 4.457 0 0 1 .038 4.5c0-2.453 2-4.448 4.452-4.448A4.457 4.457 0 0 1 8.945 4.5c0 2.452-2 4.448-4.455 4.448M-236.298 100.842a1.483 1.483 0 1 0 .002 2.965 1.483 1.483 0 0 0-.002-2.965m0 5.93a4.457 4.457 0 0 1-4.452-4.449c0-2.452 2-4.448 4.452-4.448a4.457 4.457 0 0 1 4.455 4.448c0 2.453-2 4.449-4.455 4.449"
            fill="#D8D8D8"
            fillRule="nonzero"
            mask="url(#d)"
          />
        </g>
      </g>
      <g>
        <g
          transform="translate(0 151)"
        >
          <mask
            fill="#fff"
            id="f"
          >
            <use
              xlinkHref="#e"
            />
          </mask>
          <path
            d="M4.49 3.018a1.483 1.483 0 1 0 .002 2.966 1.483 1.483 0 0 0-.002-2.966m0 5.93A4.457 4.457 0 0 1 .038 4.5c0-2.453 2-4.448 4.452-4.448A4.457 4.457 0 0 1 8.945 4.5c0 2.452-2 4.448-4.455 4.448"
            fill="#D8D8D8"
            fillRule="nonzero"
            mask="url(#f)"
          />
        </g>
      </g>
    </g>
  </g>
</svg>
`;

exports[`KeyIcon Should Render 1`] = `
<svg
  className="enable"
  height={20}
  onClick={[Function]}
  viewBox="0 0 20 20"
  width={20}
>
  <path
    d="M13.65 11A5.99 5.99 0 0 0 8 7c-3.31 0-6 2.69-6 6s2.69 6 6 6a5.99 5.99 0 0 0 5.65-4H18v4h4v-4h2v-4H13.65zM8 15c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
    fill="#D6D9DB"
    fillRule="evenodd"
    transform="rotate(-45 7.879 15.121)"
  />
</svg>
`;

exports[`Logo Should Render 1`] = `
<svg
  className="enable"
  height={21}
  onClick={[Function]}
  viewBox="0 0 154 21"
  width={154}
>
  <g
    fill="none"
    fillRule="nonzero"
  >
    <path
      d="M11.218 11.317a1.079 1.079 0 0 1-.71.23c-.28 0-.479-.06-.696-.219L3.208 4.951l-.695-.675v.003L.26 2.104V1.59C.26.96.759.462 1.387.462c.403 0 .639.126 1.006.493l8.115 7.89L18.621.958c.368-.369.604-.495 1.008-.495.628 0 1.126.5 1.126 1.127v.518l-2.252 2.175v-.006L17.12 5.618l-5.901 5.7zm7.285-4.24l2.252-2.175v14.64c0 .627-.498 1.126-1.126 1.126a1.119 1.119 0 0 1-1.126-1.126V7.077zm-15.99-.002v12.467c0 .627-.499 1.126-1.126 1.126A1.119 1.119 0 0 1 .26 19.542V4.9l2.252 2.175zM39.815.358v2.268c-3.916.515-6.946 3.872-6.946 7.927 0 4.411 3.584 7.995 7.995 7.995s7.995-3.584 7.995-7.995c0-4.085-3.074-7.462-7.033-7.938V.35a10.24 10.24 0 0 1 9.285 10.203A10.24 10.24 0 0 1 40.864 20.8a10.24 10.24 0 0 1-10.247-10.247A10.24 10.24 0 0 1 39.815.358zM60.974 4.826h2.252v3.382h2.464a2.823 2.823 0 0 0 2.812-2.812c0-1.49-1.21-2.58-2.812-2.58h-4.716V1.72c0-.627.498-1.126 1.126-1.126h3.764a4.876 4.876 0 0 1 4.89 4.89c0 1.315-.51 2.472-1.408 3.3 2.288.9 3.812 3.036 3.812 5.561a6.161 6.161 0 0 1-6.165 6.165H62.1a1.119 1.119 0 0 1-1.126-1.126V4.826zm2.252 13.49h3.623a3.985 3.985 0 0 0 3.97-3.97 3.985 3.985 0 0 0-3.97-3.97h-3.623v7.94zM141.541 4.826h2.252v3.382h2.465a2.823 2.823 0 0 0 2.812-2.812c0-1.49-1.21-2.58-2.812-2.58H141.54V1.72c0-.627.499-1.126 1.126-1.126h3.765a4.876 4.876 0 0 1 4.89 4.89c0 1.315-.511 2.472-1.408 3.3 2.287.9 3.811 3.036 3.811 5.561a6.161 6.161 0 0 1-6.164 6.165h-4.894a1.119 1.119 0 0 1-1.126-1.126V4.826zm2.252 13.49h3.623a3.985 3.985 0 0 0 3.97-3.97 3.985 3.985 0 0 0-3.97-3.97h-3.623v7.94zM85.27 1.577V19.53c0 .627-.498 1.126-1.126 1.126a1.119 1.119 0 0 1-1.126-1.126V1.577c0-.628.499-1.127 1.126-1.127.628 0 1.127.499 1.127 1.127z"
      fill="#000"
    />
    <path
      d="M129.069 20.61a.81.81 0 0 1-.052.001h-11.981a.783.783 0 0 1-.173-.019l1.07-2.193h10.058l1.078 2.21z"
      fill="#08E8DE"
    />
    <path
      d="M105.926 18.46a1.092 1.092 0 0 1 1.014 1.097c0 .63-.497 1.103-1.126 1.103h-8.398a1.122 1.122 0 0 1-1.126-1.132V18.46h9.636zm-9.636-2.011V1.582c0-.63.498-1.132 1.126-1.132.63 0 1.126.473 1.126 1.103v14.896H96.29z"
      fill="#000"
    />
    <path
      d="M114.87 20.083c-.29.512-.909.73-1.477.467-.561-.303-.785-.936-.519-1.513l8.57-17.864c.294-.62.878-.867 1.493-.867.616 0 1.2.247 1.492.865l8.574 17.872c.263.571.04 1.204-.544 1.518a1.11 1.11 0 0 1-1.238-.201.402.402 0 0 1-.086-.118l-8.197-16.874-8.068 16.715z"
      fill="#08E8DE"
    />
  </g>
</svg>
`;

exports[`UpIcon Should Render 1`] = `
<svg
  className="enable"
  height={20}
  onClick={[Function]}
  viewBox="0 0 48 48"
  width={20}
>
  <path
    d="M14.83 30.83L24 21.66l9.17 9.17L36 28 24 16 12 28z"
  />
</svg>
`;

exports[`VisibilityIcon Should Render 1`] = `
<svg
  className="enable"
  height={16}
  onClick={[Function]}
  viewBox="0 0 22 16"
  width={22}
>
  <path
    d="M11 .5C6 .5 1.73 3.61 0 8c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5C20.27 3.61 16 .5 11 .5zM11 13c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8C9.34 5 8 6.34 8 8s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
    fill="#D6D9DB"
    fillRule="evenodd"
  />
</svg>
`;
